Hazel/src/Hazel/Events/Event.h

```diff
@@ -53,20 +53,18 @@ namespace Hazel {
 
 	class EventDispatcher
 	{
-		template<typename T>
-		using EventFn = std::function<bool(T&)>;
 	public:
 		EventDispatcher(Event& event)
 			: m_Event(event)
 		{
 		}
 
-		template<typename T>
-		bool Dispatch(EventFn<T> func)
+		template<typename T, typename F>
+		bool Dispatch(const F& func)
```

Comment:
I would add a small comment (maybe even with an example) mensioning F will be deduced by the compiler, as this is something that is visible for users of Hazel.